[DOCS] Documents imbalanced class sizes and their effect on classification (#806)
elastic/stack-docs: docs/en/stack/ml/df-analytics/dfanalytics-limitations.asciidoc

destination index that don't contain a results field are not included in the 
{classanalysis}.

[float]
[[dfa-classification-imbalanced-classes]]
=== Imbalanced class sizes affect {classification} performance

If your training data is very imbalanced, then {classanalysis} may not provide 

> good predictions. The system tries to mitigate the effects of imbalanced 

Author: ```suggestion
good predictions. Training tries to mitigate the effects of imbalanced 
```

> training data by maximizing the minimum recall of any class. It means that it 

Author: ```suggestion
training data by maximizing the minimum recall of any class. For imbalanced training data, this means that it
```

> tries to get correct the highest number of examples in the minority classes. 

Reviewer: I think I'd modify these sentences slightly. Here "the system" is really training so I'd probably write something like:
"Training tries to mitigate the effects of imbalanced training data by maximizing the minimum recall of any class. For imbalanced training data, this means that it will try to balance the proportion of values it correctly labels in the minority and majority class."

> tries to get correct the highest number of examples in the minority classes. 

Author: ```suggestion
will try to balance the proportion of values it correctly labels in the minority and majority class.
```

However, this process can result in a slight degradation of the overall 

> accuracy. Try to avoid highly imbalanced situations. It is required at least 50 

Reviewer: I'd weaken these words slightly (and suggest a maximum imbalance), i.e. something like:
"We recommend having at least 50 examples of each class and a ratio of no more than 10 to 1 for the majority to minority class labels in the training data."

examples for each class in order to get better results. If your training dataset 

> is very imbalanced, fine-tune the data, for example, by downsampling the 

Reviewer: Maybe:
"If your training dataset is very imbalanced, consider downsampling the majority class, upsampling the minority class or if possible gather more data."
(I think it's worth mentioning the three possible routes to go.)
